src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputExtension.java

```diff
@@ -0,0 +1,20 @@
+package org.jenkinsci.plugins.workflow.support.steps.input;
+
+import hudson.ExtensionPoint;
+import hudson.model.Run;
+
+public interface InputExtension extends ExtensionPoint {
+    /**
+     * Callback this when an inputStep is executing.
+     * Just for notification, will ignore all exception and log them.
+     * @param inputStep instance of InputStep
+     * @param run Current building run instance
+     */
+    void input(InputStep inputStep, Run run);
+
+    /**
+     * Should be unique, for distinguish different extension.
+     * @return the name of current extension instance
+     */
+    String getName();
```

Comment:
Unclear whether this should be localizable. If not, why not just use the (simple) class name?

Reply:
The simple class name is fine. I prefer a more readable name. But if it get a null of name, it can print a simple class name there.